docs/source/developers/reviewing.rst

```diff
@@ -0,0 +1,226 @@
+
+- Avoiding worst-case degenerescence (such as memory blowups when a size
+  estimate is too imprecise) may be more critical than trying to improve
+  the common case by a small amount.
+
+Documentation
+-------------
+
+These guidelines should ideally apply to both prose documentation and
+in-code docstrings.
+
+- Look for ambiguous / sub-informative wording.  For example, "it is an error
+  if ..." is less informative than either "An error is raised if ... " or
+  "Behaviour is undefined if ..." (the first phrasing doesn't tell the
+  reader what actually *happens* on such an error).
+
+- Be mindful about spelling, grammar, expression, concision.  This may seem
+  like an obvious concern, but not all people are not accustomed to care about
+  this.
+
+- Cross-linking increases the global value of documentation.  Sphinx especially
+  has great cross-linking capabilities (including topic references, glossary
+  terms, API references), be sure to make use of them!
+
+Testing
+-------
+
+- When adding an API, all nominal cases should have test cases.  Does a function
+  allow null values? Then null values should be tested (alongside non-null
+  values, of course). Does a function allow different input types? etc.
+
+- If some aspect of a functionality is delicate (either by definition or
+  as an implementation detail), it should be tested.
+
+- Corner cases should be exercised, especially in low-level implementation
+  languages such as C++.  Examples: empty arrays, zero-chunk arrays, arrays
+  with only nulls, etc.
+
+- Stress tests can be useful, for example to uncover synchronizations bugs
+  if non-trivial parallelization is being added, or to validate a computational
+  argument against a slow and straightforward reference implementation.
+
+- A mitigating concern, however, is the overall cost of running the test
+  suite.  Continuous Integration (CI) runtimes can be painfully long and
+  we should be wary of increasing them too much.  Sometimes it is
+  worthwhile to fine-tune testing parameters to balance the usefulness
+  of tests against the cost of running them (especially where stress tests
+  are involved, since they tend to imply execution over large datasets).
+
+
+Social aspects
+==============
+
+* Reviewing is a communication between the contributor and the reviewer.
+  Avoid letting questions or comments unanswered for too long ("too long"
+  is of course very subjective, but two weeks can be a reasonable
+  heuristic).  If you cannot allocate time soon, do say it explicitly.
+  If you don't have the answer to a question, do say it explicitly.
+  Saying "I don't have time immediately but I will come back later,
+  feel free to ping if I seem to have forgotten" or "Sorry, I am out of depth
+  here" is always better than saying nothing and letting the interlocutor
+  wondering.
+
+* If you know someone who may help on a blocking issue and past experience
+  suggests they can be available for that, feel free to add them to the
```

Comment:
"available" here seems orthogonal to expertise shown through past experience. Should we phrase this about both expertise and that the ping is a way to assess availability?

Reply:
I know I am frequently hesitant to `@` someone, but I should more frequently and they can say either "sorry I don't have time" or "oh this is the answer..."

Reply:
Hmm, who about:

"If you know someone who has the competence to help on a blocking issue and past experience suggests they may be willing to do so, feel free to add them to the discussion (for example by gently pinging their Github handle)."